airflow/executors/base_executor.py

```diff
@@ -221,30 +222,59 @@ def heartbeat(self) -> None:
         num_running_tasks = len(self.running)
         num_queued_tasks = len(self.queued_tasks)
 
-        self.log.debug("%s running task instances", num_running_tasks)
-        self.log.debug("%s in queue", num_queued_tasks)
-        self.log.debug("%s open slots", open_slots)
+        self._emit_metrics(open_slots, num_running_tasks, num_queued_tasks)
+        self.trigger_tasks(open_slots)
+
+        # Calling child class sync method
+        self.log.debug("Calling the %s sync method", self.__class__)
+        self.sync()
+
+        self.trigger_tasks(open_slots)
+
+        # Calling child class sync method
+        self.log.debug("Calling the %s sync method", self.__class__)
+        self.sync()
```

Comment:
I think those lines are duplicated by mistake

Reply:
Not sure which lines you're referring to. I just took all the lines that are related to metrics, and put them in `_emit_metrics`. The rest are left as is from the original code.

Reply:
I talked about those lines: [1](https://github.com/apache/airflow/blob/cef21508289ff3f2be5a382c66ee6114c61fa7c9/airflow/executors/base_executor.py#L226-L230) [2](https://github.com/apache/airflow/blob/cef21508289ff3f2be5a382c66ee6114c61fa7c9/airflow/executors/base_executor.py#L232-L236)

and those line occurs only once in the main version [here](https://github.com/apache/airflow/blob/main/airflow/executors/base_executor.py#L242-L246)

Reply:
oh wow, I totally missed that. Fixed thanks